New: Parsing POL as Polish and EN as English

pcela opened the pull request:
Adding the phrase "POL" to indicate the Polish language and the phrase "en" to indicate English.

#### Database Migration
NO

#### Description
Adding the phrase "POL" to indicate the Polish language and the phrase "en" to indicate English. This is intended to identify files marked with both "PL" and "POL" for Polish, as well as "en" and "eng" for English.


An example release where it does not detect all languages ​​is: "Godzilla King of the Monsters 2019 Eng Fre Ger Ita Spa Cze Hun Pol Rus Hin Tam Tel Tha Tur 2160p BluRay Remux DV HDR HEVC Atmos-SGF". Languages ​​have been marked here by ISO codes 639-2/B.

Of course it still does not recognize all languages, but I focused on European languages.

pcela committed:
Update LanguageParser.cs

Adding the phrase "POL" to indicate the Polish language and the phrase "en" to indicate English.

bakerboy448 reviewed (changes requested):
- ci failed
- Need to add new tests for POL and additional test cases for EN for English

mynameisbogdan reviewed (changes requested):
Please add test cases for your changes in `LanguageParserFixture`.

pcela committed:
Adding the phrase "POL" to indicate the Polish language and the phrase "en" to indicate English. This is intended to identify files marked with both "PL" and "POL" for Polish, as well as "en" and "eng" for English v2. Update LanguageParser.cs

Removed redundant parenthesis “(”.

pcela commented on src/NzbDrone.Core/Parser/LanguageParser.cs:
```diff
@@ -38,7 +38,7 @@ public static class LanguageParser
         private static readonly Regex CaseSensitiveLanguageRegex = new Regex(@"(?:(?i)(?<!SUB[\W|_|^]))(?:(?<lithuanian>\bLT\b)|
                                                                                                           (?<czech>\bCZ\b)|
-                                                                                                          (?<polish>\bPL\b)|
+                                                                                                          (?<polish>\b(?:(PL|POL)\b)|
```
I removed the redundant parenthesis “(”. Now it should be OK.

pcela committed:
Update LanguageParser.cs

Addition of earlier upgrade to detect languages:
- German - marked by DE, DEU, Deu and Ger
- Polish - marked by PL, POL or Pol,
- Czech - marked by CZ, CZE or Cze,
- Turkish - marked by TR, TUR or Tur.

This should provide detection of languages ​​by ISO code 639-1, 639-2 and 639-3.

An example release where it does not detect all languages ​​is: "Godzilla King of the Monsters 2019 Eng Fre Ger Ita Spa Cze Hun Pol Rus Hin Tam Tel Tha Tur 2160p BluRay Remux DV HDR HEVC Atmos-SGF". The languages ​​here are marked with ISO codes 639-2/B.

Of course it still does not recognize all languages, but I focused on European languages.

pcela committed:
Update LanguageParser.cs

Addition of bracket. Accidentally removed.

pcela committed:
Update LanguageParser.cs

Remove "|"in line 41.

bakerboy448 commented:
Closing appears abandoned by OP and is no where near any state that would be merged. Requested changes never made.

Feel free to PR a working version at a future date
